platform-sdk/swirlds-merkledb/src/main/java/com/swirlds/merkledb/collections/LongListDisk.java

```diff
@@ -194,7 +194,7 @@ private static void fillBufferWithZeroes(ByteBuffer transferBuffer) {
 
     private ByteBuffer initOrGetTransferBuffer() {
         ByteBuffer buffer = TRANSFER_BUFFER_THREAD_LOCAL.get();
-        if (buffer == null) {
+        if (buffer == null || buffer.capacity() != memoryChunkSize) {
```

Comment:
It can be done even better. `buffer.capacity()` doesn't have to be equal to `memoryChunkSize`, it has to be greater or equal. When it's greater, we need to make sure buffer limit is never reset, e.g. `buffer.clear()` is not called like it is today in `LongListDisk.fillBufferWithZeroes()`.

So the proposal is:
* Change the condition above to be greater or equal rather than equal
* Set buffer limit to memoryChunkSize in the end of `initOrGetTransferBuffer()`
* Inspect all calls to `buffer.clear()` and replace them with `setPosition(0)` + `setLimit(memoryChunkSize)`

Reply:
https://github.com/hashgraph/hedera-services/pull/16962/commits/9bda90b0ab51806f2342aafdcf08f283b2b41b61

Reply:
Actually, I merged `develop`, and it already contains the fix as you describe it, which came from https://github.com/hashgraph/hedera-services/pull/16962, so this PR will contain only a test.

Reply:
@artemananiev actually, I merged `develop`, and it already contains the fix as you describe it, which came from https://github.com/hashgraph/hedera-services/pull/16962, so this PR will contain only a test.